tests/unittests/parsing/LexerTests.cpp

```diff
@@ -1186,6 +1186,21 @@ TEST_CASE("Token with lots of trivia") {
     CHECK(trivia.getExplicitLocation()->offset() == 5);
 }
 
+TEST_CASE("Directive trivia location") {
+    auto& text = "//bar\n`define FOO bar";
+    Token token = lexToken(text);
+
+    CHECK(token.kind == TokenKind::EndOfFile);
+    REQUIRE(token.trivia().size() == 1);
+
+    Trivia t = token.trivia()[0];
+    CHECK(t.kind == TriviaKind::Directive);
+    REQUIRE(t.syntax()->kind == SyntaxKind::DefineDirective);
+    CHECK(t.getExplicitLocation()->offset() == 0);
```

Comment:
Ths check fails, the returned location points to the start of the directive and doesn't count the nested trivia that precede it.